pkg/controller/backup_sync_controller.go

```diff
@@ -272,6 +272,10 @@ func (b *backupSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			for _, snapCont := range snapConts {
 				// TODO: Reset ResourceVersion prior to persisting VolumeSnapshotContents
 				snapCont.ResourceVersion = ""
+				// For creating static VSContent, we should put snapshothandle in source rather than volume handle.
+				// Because if VSContent syncs to a different cluster, having volumeHandle will force rePUTs on the snapshot
+				snapCont.Spec.Source.SnapshotHandle = snapCont.Status.SnapshotHandle
+				snapCont.Spec.Source.VolumeHandle = nil
```

Comment:
Will the `SnapshotHandle` value be unique globally ? Or does it depend on:
- whether the storage across clusters is shared ?
- the csi driver of the storage provider ?

Reply:
snapshotHandle in my experience is the unique identifier at the cloud provider level.

As long as it's the same cloud provider, same provisioner, then same snapshothandle = same unique content globally.

Reply:
snapshotHandle for aws provisioner would be the same value as SnapshotId of an ec2 EBS snapshot
[`"SnapshotId": "snap-066877671789bd71b"
`](https://docs.aws.amazon.com/cli/latest/reference/ec2/create-snapshot.html#:~:text=%22SnapshotId%22%3A%20%22snap%2D066877671789bd71b%22) which for the same account credentials should be unique globally regardless of cluster.

Reply:
My understanding is also that same that it will mostly be unique across
Using a GUID in any naming ensures that.

In case of Azure it's a fully qualified URL with subscription, RG and snapshot name.

But again any GUID should ensure uniqueness.

Reply:
@shubham-pampattiwar any other thoughts?

Reply:
If possible, Could you provide a link to the change/PR that is causing the rePUTs on the snapshot ?

Reply:
tagged on slack